File: Algorithms/p377_Combination_Sum_IV_2.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) {
        vector<int> table(target+1,0);
        table[0] = 1;
        for(int i=1;i<=target;++i){
            for(int v : nums){
                if(i>=v){
                    table[i] += table[i-v];
                }
            }
        }
        return table[target];
    }
};

File: Algorithms/p377_Combination_Sum_IV_2.py
class Solution(object):
    def combinationSum4(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        table = [0]*(target+1)
        table[0] = 1
        for i in range(1,target+1):
            for v in nums:
                if i>=v:
                    table[i] += table[i-v]
        return table[target]
